perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ExpirationNotifScheduler.java

```diff
@@ -357,6 +378,82 @@ private void checkVoMembersState(List<Vo> vos) throws WrongAttributeAssignmentEx
 		expireSponsorships();
 	}
 
+	/**
+	 * Checks all applications for given vos and if some application is expired (according to expiration rules set by
+	 * VO Manager), then rejects this application.
+	 *
+	 * @param vos virtual organizations
+	 * @throws PerunException
+	 */
+	private void voApplicationsAutoRejection(List<Vo> vos) throws PerunException {
+		List<String> states = new ArrayList<>();
+		states.add("NEW");
+		states.add("VERIFIED");
+
+		for (Vo vo : vos) {
+			Attribute expiration = perun.getAttributesManagerBl().getAttribute(sess, vo, A_VO_APP_EXP_RULES);
+			if (expiration != null) {
```

Comment:
I think the attribute will always be not null. You should instead check its value:
```suggestion
			if (expiration.getValue() != null) {
```

Reply:
Changed